### PR TITLE
Compare Tavily's scraping vs. our scraping

### PR DESCRIPTION
**Pull request by kongzii**
Demo allows to do a query and then compare scraps side by side:

Screenshot by Dropbox Capture

Going to do a summary in the comment.

### CODE REVIEW

**Comment by kongzii** [comment]
Query: What will be the first ten countries to do widespread embryo selection for intelligence?

Tavily 8k words vs. we 10k words. However, page's main content starts with "Embryo selection within in vitro fertilization", and Tavily's scrap is much closer to this:

Screenshot by Dropbox Capture

Then, page's main content ends with "randomized controlled trials is the only valid evaluation." and that's more or less where Tavily is ending:

Screenshot by Dropbox Capture

So we scraped more, but uninteresting content for the topic. Tavily cut out citations and some other stuff.

**Comment by kongzii** [comment]
Query: What will be the first ten countries to do widespread embryo selection for intelligence?

URL: https://medicalxpress.com/news/2022-12-harnessing-artificial-intelligence-technology-ivf.html

Tavily 1.3k words, vs. we 2.1k words. Again the same scenario as before, we do scrap more, but non-relevant stuff from the website.

For example, we scrape these ads for other articles on the right too

Screenshot by Dropbox Capture

**Comment by kongzii** [comment]
Query: What's Gemini

URL: https://www.horoscope.com/zodiac-signs/gemini

Similar as before, except that now, Tavily excluded some potential helpful information near the end, the highlighted bullets:

Screenshot by Dropbox Capture

**Comment by kongzii** [comment]
Query: Conservative activist Charlie Kirk helped oust Ronna McDaniel at the RNC. Now the knives are out for him.

URL: https://news.yahoo.com/conservative-activist-charlie-kirk-helped-130000090.html

Funny enough, we scraped some text that I can't even find on the site :-D Including text like

```
Something went wrong.Please try again later.More content belowOops!Something went wrong.Please try again later.More content belowOops!Something went wrong.Please try again later.More content belowOops!Something went wrong.Please try again later.More content belowOops!Something went wrong.Please try again later
```

But Tavily didn't.

**Comment by kongzii** [comment]
Query: How to train chiuauas 

URL: https://thehappypuppysite.com/how-to-train-a-chihuahua/

Our scaper doesn't do anything there, while Tavily's gets through:

Screenshot by Dropbox Capture

**Comment by kongzii** [comment]
TLDR: Tavily can get through some blocks that simple requests.get won't do. They also try to extract only helpful part of the page, although that sometimes fails. I added optional `use_tavily_raw_content` and will run a benchmark in the other PR. We can decide based on that to enable or not it by default.

File: evo_researcher/functions/parallelism.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from concurrent.futures import Executor
 from concurrent.futures.thread import ThreadPoolExecutor
 
-THREADPOOL = ThreadPoolExecutor(int(os.getenv("THREADPOOL_N_THREADS", 50)))
+THREADPOOL = ThreadPoolExecutor(int(os.getenv("THREADPOOL_N_THREADS", 5)))
```
Comment (kongzii) [inline evo_researcher/functions/parallelism.py:7]: Limiting to just 5, to be the same default as in other parallel stuff.

File: evo_researcher/functions/web_scrape.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def fetch_html(url: str, timeout: int) -> Response:
     return response
 
 
-def web_scrape(url: str, timeout: int = 10000) -> tuple[str, str]:
+def web_scrape(url: str, timeout: int = 10000) -> str:
```
Comment (kongzii) [inline evo_researcher/functions/web_scrape.py:20]: Thanks to the change above, no need to return the original URL here.

**Review by evangriffiths — APPROVED** [review]
Nice findings!
